features/workflow.feature

```diff
@@ -16,125 +16,122 @@ Feature: Workflow
+            | PARALLEL1 | SIMPLE    |
+            | PARALLEL2 | SIMPLE    |
+        When a new commit is pushed to "parallel" branch
+        And the "SIMPLE" job is triggered
+        And the "SIMPLE" build succeeded
+        Then the "PARALLEL1" job is triggered from "SIMPLE"
+        And the "PARALLEL2" job is triggered from "SIMPLE"
+        And that "PARALLEL1" build uses the same SHA as the "SIMPLE" build
+        And that "PARALLEL2" build uses the same SHA as the "SIMPLE" build
+        And the "STAGING" job is not triggered
+        And that "REGEX" build uses the same SHA as the "SIMPLE" build
 
     Scenario: Join
-        Given an existing pipeline with the workflow:
-            | job   | requires |
-            | FOO   | ~commit  |
-            | BAR   | FOO      |
-            | BAZ   | FOO      |
-            | QUX   | BAR, BAZ |
-        When the "FOO" job is started
-        And the "FOO" build succeeded
-        And the "BAR" job is started
-        And the "BAZ" job is started
-        Then the "QUX" job is not started
-        And the "BAR" build succeeded
-        And the "BAZ" build succeeded
-        Then the "QUX" job is started
-        And that "QUX" build uses the same SHA as the "FOO" build
+        Given an existing pipeline on "join" branch with the workflow jobs:
+            | job       | requires              |
+            | SIMPLE    | ~commit               |
+            | PARALLEL1 | SIMPLE                |
+            | PARALLEL2 | SIMPLE                |
+            | JOIN      | PARALLEL1, PARALLEL2  |
+        When a new commit is pushed to "join" branch
+        And the "SIMPLE" job is triggered
+        And the "SIMPLE" build succeeded
+        And the "PARALLEL1" job is triggered from "SIMPLE"
+        And the "PARALLEL1" build succeeded
+        And the "PARALLEL2" job is triggered from "SIMPLE"
+        And the "PARALLEL2" build succeeded
+        Then the "JOIN" job is triggered from "PARALLEL1" and "PARALLEL2"
+        And that "JOIN" build uses the same SHA as the "SIMPLE" build
 
     Scenario: Branch filtering (the master branch is committed)
-        Given an existing pipeline with the workflow:
-            | job   | requires        |
-            | FOO   | ~commit         |
-            | BAR   | ~commit:staging |
-            | BAZ   | ~commit:/^.*$/  |
-        When a new commit is pushed
-        And it is on the "master" branch
-        Then the "FOO" job is started
-        And the "BAR" job is not started
-        And the "BAZ" job is started
-        And that "BAZ" build uses the same SHA as the "FOO" build
+        Given an existing pipeline on "master" branch with the workflow jobs:
+            | job       | requires          |
+            | SIMPLE    | ~commit           |
+            | STAGING   | ~commit:staging   |
+            | REGEX     | ~commit:/^.*$/    |
+        When a new commit is pushed to "master" branch
+        Then the "SIMPLE" job is triggered
```

Comment:
Don't you need `And the "STAGING" job is not started`?

Reply:
I should add the step, thank you!